truex-ios-mobile-reference-app/VideoPlayerViewController.m

```diff
@@ -105,87 +106,110 @@ - (void)adBreakStarted {
     [self.player play];
 }
 
+// [4]
 - (void)onNoAdsAvailable {
+    // true[X] - TruexAdRenderer has no ads ready to present, resume with standard ads
     NSLog(@"truex: onNoAdsAvailable");
     [self resetActiveAdRenderer];
     [self.player play];
 }
 
+// [3] - Respond to onAdFreePod
 - (void)onAdFreePod {
+    // true[X] - User has met engagement requirements, skips past remaining pod ads
     NSLog(@"truex: onAdFreePod");
-    if (_resumeTime != -1) {
-        // Custom resume logic for snap back
+    if (_resumeTime == -1) {
+        // true[X] - Skipping the whole ad break here as user earn credit from true[X]
+        [self seekOverCurrentAdBreak];
+    } else {
+        // Custom snap back logic, skipping ad break and send user back to their original position
         [self.player seekToTime:CMTimeMake(_resumeTime, 1)];
         _resumeTime = -1;
-    } else {
-        [self seekOverCurrentAdBreak];
     }
     [self helperEndAdBreak];
 }
 
+// [5] - Other delegate method
 - (void)onPopupWebsite:(NSString *)url {
+    // true[X] - User wants to open an external link in the true[X] ad
+
     NSLog(@"truex: onPopupWebsite: %@", url);
     // Open the URL in Safari
     // [[UIApplication sharedApplication] openURL:[NSURL URLWithString: url] options:@{} completionHandler:nil];
     
-    // Or open with your existing in app webview
+    // Or, open with the existing in-app webview
     UIStoryboard* storyBoard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
     WebViewViewController* newViewController = [storyBoard instantiateViewControllerWithIdentifier:@"webviewVC"];
     newViewController.url = [NSURL URLWithString:url];
     newViewController.modalPresentationStyle = UIModalPresentationPopover;
     __weak typeof(self) weakSelf = self;
     newViewController.onDismiss = ^(void) {
+        // true[X] - You will need to pause and remume the true[X] Ad Renderer
         [weakSelf.activeAdRenderer resume];
     };
     [self.activeAdRenderer pause];
     [self presentViewController:newViewController animated:YES completion:nil];
 }
 
-// @optional
+// @optional true[X] delegate methods
+// [5]
 -(void) onOptIn:(NSString*)campaignName adId:(NSInteger)adId {
+    // true[X] - This event is triggered when a user decides opt-in to the true[X] interactive ad
     NSLog(@"truex: onOptIn: %@, %li", campaignName, (long)adId);
-    
 }
 
+// [5]
 -(void) onOptOut:(BOOL)userInitiated {
+    // true[X] - User has opted out of true[X] engagement, show standard ads
     NSLog(@"truex: userInitiated: %@", userInitiated? @"true": @"false");
 }
 
+// [5]
 -(void) onSkipCardShown {
+    // true[X] - TruexAdRenderer displayed a Skip Card
     NSLog(@"truex: onSkipCardShown");
 }
 
+// [5]
 -(void) onUserCancel {
+    // true[X] - This event will fire when a user backs out of the true[X] interactive ad unit after having opted in.
     NSLog(@"truex: onUserCancel");
 }
 
-// MARK: - Helper Functions
+// MARK: - Helper Functions / Fake Ad Manager
```

Comment:
Would "Fake Ad Server Call" be more specific here?

Reply:
Also peaked below this at   `// Or use the hardcoded copy`.

I like what we're going for here. The idea that we're showing an alternative code path. Can we maybe make it clearer that the next two commented out lines can be used instead of lines before them?